js/renderer.js

```diff
@@ -106,7 +122,12 @@ class Renderer {
         this.ctx.save();
         this.ctx.translate(this.camera.translateX(sprite.getPointOfRotationX()), this.camera.translateY(sprite.getPointOfRotationY()));
         this.ctx.rotate(sprite.rotation);
-        this.ctx.drawImage(img, -Math.round(sprite.rotationOffsetX), -Math.round(sprite.rotationOffsetY), sprite.width, sprite.height);
+        const zoomFactor = this.camera.zoom;
+        const x = -Math.round(sprite.rotationOffsetX * zoomFactor);
+        const y = -Math.round(sprite.rotationOffsetY * zoomFactor);
+        const w = Math.round(sprite.width * zoomFactor);
+        const h = Math.round(sprite.height * zoomFactor);
```

Comment:
Perhaps make this into its own method? Although you did say this was a hack until everything has its own sprite.

Reply:
i definitely considered it, but both methods that will eventually draw images to the canvas (renderSprite and renderRotatedSprite) do it slightly differently, so i couldnt think of a clean way to extract this logic